src/test/java/com/snowflake/kafka/connector/SmtIT.java

```diff
@@ -0,0 +1,84 @@
+package com.snowflake.kafka.connector;
+
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.BUFFER_FLUSH_TIME_SEC;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.INGESTION_METHOD_OPT;
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.NAME;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.KEY_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.TASKS_MAX_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.TRANSFORMS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.VALUE_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.sink.SinkConnector.TOPICS_CONFIG;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import com.snowflake.kafka.connector.internal.TestUtils;
+import com.snowflake.kafka.connector.internal.streaming.FakeStreamingClientHandler;
+import com.snowflake.kafka.connector.internal.streaming.IngestionMethodConfig;
+import com.snowflake.kafka.connector.internal.streaming.StreamingClientProvider;
+import java.time.Duration;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.storage.StringConverter;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class SmtIT extends ConnectClusterBaseIT {
+
+  private static final String SMT_TOPIC = "SMT_TOPIC";
+  private static final String SMT_CONNECTOR = "SMT_CONNECTOR";
+  private static final FakeStreamingClientHandler fakeStreamingClientHandler =
+      new FakeStreamingClientHandler();
+
+  @BeforeAll
+  public void createConnector() {
+    StreamingClientProvider.overrideStreamingClientHandler(fakeStreamingClientHandler);
```

Comment:
Ideally we also should to revert this override back as other tests can still depend on a real implementation.
To strengthen test isolation will it make sense to create a new `FakeStreamingClientHandler` for each test?

Also, let's move this logic into `ConnectClusterBaseIT`, as I expect every test will override the handler.

Reply:
done